### PR TITLE
feat: UI setup for feedback component hidden behind feature flag

### PR DESCRIPTION
**Pull request by ianjon3s**
## What

- New UI components built by Ian as per [designs](https://www.figma.com/file/bnUUrsVRG6qPwDkTmVKACI/PlanX-Design?type=design&node-id=12766-9275&mode=design) and [prototype](https://www.figma.com/proto/bnUUrsVRG6qPwDkTmVKACI/PlanX-Design?page-id=12766%3A9275&type=design&node-id=13794-32506&viewport=-1569%2C471%2C0.7&t=hHNmEtfB7iz7Lk6T-1&scaling=min-zoom&starting-point-node-id=13794%3A32506&mode=design)
- Reorganisation of the components to allow them to be hidden behind a feature flag

## Why

- Useful to split up the PR for comprehensibility

## Testing 

- Demo service: https://2632.planx.pizza/testing/feedback-ui/preview
- Check that the UI defaults to current designs (barring background difference on Help Text)
- Toggle the feature flag: `window.featureFlags.toggle("SHOW_INTERNAL_FEEDBACK")`
- View the new UI 

## Screen recording

- https://github.com/theopensystemslab/planx-new/assets/36415632/6a2779c3-bf1d-4e88-9272-91430bfcf69a

### CODE REVIEW

**Comment by jessicamcinchak** [comment]
[Design & implement in-house feedback feature & retire Feedback Fish](https://trello.com/c/8pmX6wmZ/2545-design-implement-in-house-feedback-feature-retire-feedback-fish)

File: hasura.planx.uk/migrations/1705407249614_squashed/up.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+CREATE TABLE "public"."feedback_type_enum" ("value" text NOT NULL, "comment" text NOT NULL, PRIMARY KEY ("value") );COMMENT ON TABLE "public"."feedback_type_enum" IS E'Store the different types of feedback we gather from user';
+
+INSERT INTO "public"."feedback_type_enum"("value", "comment") VALUES (E'issue', E' A user reporting an issue with a service');
```
Comment (DafyddLlyr) [inline hasura.planx.uk/migrations/1705407249614_squashed/up.sql]:
```suggestion
INSERT INTO "public"."feedback_type_enum"("value", "comment") VALUES (E'issue', E'A user reporting an issue with a service');
```

Comment (Mike-Heneghan) [reply hasura.planx.uk/migrations/1705407249614_squashed/up.sql]: I've made this change on another branch so this comment hasn't been lost 😌

File: editor.planx.uk/src/@planx/components/shared/Preview/MoreInfo.tsx [hunk 2]
```diff
@@ -59,6 +62,8 @@ const CloseButton = styled("div")(({ theme }) => ({
   color: theme.palette.text.primary,
 }));
 
+const isUsingFeatureFlag = () => hasFeatureFlag("SHOW_INTERNAL_FEEDBACK");
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/shared/Preview/MoreInfo.tsx:65]: I _think_ we'd save a few renders by just making this a static value?

`const isUsingFeatureFlag = hasFeatureFlag("SHOW_INTERNAL_FEEDBACK");`

File: editor.planx.uk/src/ui/public/FeedbackDisclaimer.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import Typography from "@mui/material/Typography";
+import React from "react";
+
+export default function FeedbackDisclaimer(): FCReturn {
+  return (
+    <Box>
+      <Typography variant="body2">
+        Please do not include any personal or financial information in your
+        feedback. If you choose to do so we will process your personal data
+        according to our <Link>privacy policy</Link>.
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/ui/public/FeedbackDisclaimer.tsx:12]: We'll need to make a note to come back and populate this link. Maybe let's make it visually obvious with `<Link href="#">privacy policy (TODO)</Link>`

**Review by DafyddLlyr — APPROVED** [review]
A few very small suggestions but looking great! 👍

File: editor.planx.uk/src/@planx/components/shared/Preview/MoreInfo.tsx (hunk 2 above)
Comment (Mike-Heneghan) [reply editor.planx.uk/src/@planx/components/shared/Preview/MoreInfo.tsx:65]: That's an interesting one, I think I just copied the pattern from the other places where a feature flag is used and didn't stop to think about it 😅 

I'll experiment with making it static to see whether it's like this for a reason or maybe just causing unnecessary renders.

Comment (Mike-Heneghan) [reply editor.planx.uk/src/@planx/components/shared/Preview/MoreInfo.tsx]: Experimenting with this it seems the static version is grand, when the flag is toggled it triggers the refresh so it seems to be working as intended. 

Fixed: adcb8c2c691d9c78a0655d1c0c7aaf7c3b98e189

File: editor.planx.uk/src/ui/public/FeedbackDisclaimer.tsx (hunk 3 above)
Comment (Mike-Heneghan) [reply editor.planx.uk/src/ui/public/FeedbackDisclaimer.tsx]: Fixed here: aa17f630398e085f289a8de8abe2a37b9dc22889

**Comment by Mike-Heneghan** [comment]
[Design & implement in-house feedback feature & retire Feedback Fish](https://trello.com/c/8pmX6wmZ/2545-design-implement-in-house-feedback-feature-retire-feedback-fish)

**Pull request merged**